From de3cc4e26cca5f7c53c7621754b8cac891ce85e4 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Sat, 28 Mar 2026 13:09:02 +0100
Subject: [PATCH] Disable building documentation

---
 gcc/Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/Makefile.in b/gcc/Makefile.in
index 4d10ffb..b16f2e2 100644
--- a/gcc/Makefile.in
+++ b/gcc/Makefile.in
@@ -2480,8 +2480,8 @@ stmp-fixproto: fixhdr.ready fixproto stmp-int-hdrs
 
 docdir = $(srcdir)/doc
 
-doc: $(BUILD_INFO) $(GENERATED_MANPAGES) gccbug
-info: $(docdir)/cpp.info $(docdir)/gcc.info $(docdir)/gccint.info $(docdir)/gccinstall.info lang.info $(docdir)/cppinternals.info
+doc: 
+info: 
 
 TEXI_CPP_FILES = $(docdir)/cpp.texi $(docdir)/include/fdl.texi \
   $(docdir)/cppenv.texi $(docdir)/cppopts.texi
@@ -2755,7 +2755,7 @@ maintainer-clean:
 # Install the driver last so that the window when things are
 # broken is small.
 install: install-common $(INSTALL_HEADERS) $(INSTALL_LIBGCC) \
-    install-cpp install-man install-info intl.install install-@POSUB@ \
+    install-cpp intl.install install-@POSUB@ \
     lang.install-normal install-driver
 
 # Handle cpp installation.
-- 
2.53.0

