From 2ba3f8416ea383f30370b01db17eba7bc4699890 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Fri, 27 Mar 2026 22:31:39 +0100
Subject: [PATCH] Prevent use of *_unlocked() functions and disable inclusion
 of <malloc.h>

---
 gcc/system.h | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/gcc/system.h b/gcc/system.h
index e19de3f8..a8c64da8 100644
--- a/gcc/system.h
+++ b/gcc/system.h
@@ -51,7 +51,7 @@ Software Foundation, 59 Temple Place - Suite 330, Boston, MA
    functions can cause the compiler to be significantly slower under
    I/O bound conditions (such as -g -O0 on very large source files).
 
-   HAVE_DECL_PUTC_UNLOCKED actually indicates whether or not the stdio
+   HAVE_DECL_PUTC_UNLOCKED____ actually indicates whether or not the stdio
    code is multi-thread safe by default.  If it is set to 0, then do
    not worry about using the _unlocked functions.
 
@@ -59,37 +59,37 @@ Software Foundation, 59 Temple Place - Suite 330, Boston, MA
    extensions and need to be prototyped by hand (since we do not
    define _GNU_SOURCE).  */
 
-#if defined HAVE_DECL_PUTC_UNLOCKED && HAVE_DECL_PUTC_UNLOCKED
+#if defined HAVE_DECL_PUTC_UNLOCKED__ && HAVE_DECL_PUTC_UNLOCKED__
 
-# ifdef HAVE_PUTC_UNLOCKED
+# ifdef HAVE_PUTC_UNLOCKED__
 #  undef putc
 #  define putc(C, Stream) putc_unlocked (C, Stream)
 # endif
-# ifdef HAVE_FPUTC_UNLOCKED
+# ifdef HAVE_FPUTC_UNLOCKED__
 #  undef fputc
 #  define fputc(C, Stream) fputc_unlocked (C, Stream)
 # endif
 
-# ifdef HAVE_FPUTS_UNLOCKED
+# ifdef HAVE_FPUTS_UNLOCKED__
 #  undef fputs
 #  define fputs(String, Stream) fputs_unlocked (String, Stream)
-#  if defined (HAVE_DECL_FPUTS_UNLOCKED) && !HAVE_DECL_FPUTS_UNLOCKED
+#  if defined (HAVE_DECL_FPUTS_UNLOCKED__) && !HAVE_DECL_FPUTS_UNLOCKED__
 extern int fputs_unlocked (const char *, FILE *);
 #  endif
 # endif
-# ifdef HAVE_FWRITE_UNLOCKED
+# ifdef HAVE_FWRITE_UNLOCKED__
 #  undef fwrite
 #  define fwrite(Ptr, Size, N, Stream) fwrite_unlocked (Ptr, Size, N, Stream)
-#  if defined (HAVE_DECL_FWRITE_UNLOCKED) && !HAVE_DECL_FWRITE_UNLOCKED
+#  if defined (HAVE_DECL_FWRITE_UNLOCKED__) && !HAVE_DECL_FWRITE_UNLOCKED__
 extern int fwrite_unlocked (const void *, size_t, size_t, FILE *);
 #  endif
 # endif
-# ifdef HAVE_FPRINTF_UNLOCKED
+# ifdef HAVE_FPRINTF_UNLOCKED__
 #  undef fprintf
 /* We can't use a function-like macro here because we don't know if
    we have varargs macros.  */
 #  define fprintf fprintf_unlocked
-#  if defined (HAVE_DECL_FPRINTF_UNLOCKED) && !HAVE_DECL_FPRINTF_UNLOCKED
+#  if defined (HAVE_DECL_FPRINTF_UNLOCKED__) && !HAVE_DECL_FPRINTF_UNLOCKED__
 extern int fprintf_unlocked (FILE *, const char *, ...);
 #  endif
 # endif
@@ -315,7 +315,7 @@ extern void *sbrk (int);
 extern char *strstr (const char *, const char *);
 #endif
 
-#ifdef HAVE_MALLOC_H
+#ifdef HAVE_MALLOC_H__
 #include <malloc.h>
 #endif
 
-- 
2.53.0

