From 4b872a13157137b571719cf89c26ad8e7d29d30b Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Fri, 20 Mar 2026 07:02:11 -0300
Subject: [PATCH] Remove versioned SONAME from libmpc

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index f9fd0b3..6e5bcb0 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -394,7 +394,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 lib_LTLIBRARIES = libmpc.la
-libmpc_la_LDFLAGS = $(MPC_LDFLAGS) -version-info 7:1:4
+libmpc_la_LDFLAGS = $(MPC_LDFLAGS) -avoid-version
 libmpc_la_SOURCES = mpc-impl.h abs.c acos.c acosh.c add.c add_fr.c         \
   add_si.c add_ui.c agm.c arg.c asin.c asinh.c atan.c atanh.c clear.c      \
   cmp.c cmp_abs.c cmp_si_si.c conj.c cos.c cosh.c                          \
-- 
2.52.0

