From 09b566d43bf086f52c3e4ba832698ff9e5500209 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Sun, 8 Feb 2026 04:02:34 -0300
Subject: [PATCH] Remove hardcoded RPATH and versioned SONAME from libmpfr

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 9f64eed..c8ae16e 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -623,7 +623,7 @@ libmpfr_la_LIBADD = @LIBOBJS@ $(am__append_2)
 #        4.0.x    6:x:0
 #        4.1.x    7:x:1
 #        4.2.x    8:x:2
-libmpfr_la_LDFLAGS = $(MPFR_LDFLAGS) $(LIBMPFR_LDFLAGS) -version-info 8:0:2
+libmpfr_la_LDFLAGS = $(MPFR_LDFLAGS) $(LIBMPFR_LDFLAGS) -avoid-version
 GET_PATCHES_SH = $(top_srcdir)/tools/get_patches.sh
 PATCHES = $(top_srcdir)/PATCHES
 
-- 
2.52.0

