From 0f996c6fe6b3fb82afe1dd70e528d6ef05ed2dab Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Thu, 2 Oct 2025 04:09:31 -0300
Subject: [PATCH 01/11] Turn -Wimplicit-function-declaration back into an
 warning

---
 gcc/c/c-decl.cc | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/gcc/c/c-decl.cc b/gcc/c/c-decl.cc
index 7e4c7c2ac..d2ef91d9d 100644
--- a/gcc/c/c-decl.cc
+++ b/gcc/c/c-decl.cc
@@ -3565,7 +3565,7 @@ pushdecl (tree x)
    or NULL_TREE.  */
 
 static void
-implicit_decl_permerror (location_t loc, tree id, tree olddecl)
+implicit_decl_warning (location_t loc, tree id, tree olddecl)
 {
   if (!warn_implicit_function_declaration)
     return;
@@ -3582,13 +3582,13 @@ implicit_decl_permerror (location_t loc, tree id, tree olddecl)
 	{
 	  gcc_rich_location richloc (loc);
 	  richloc.add_fixit_replace (suggestion);
-	  warned = permerror_opt (&richloc, OPT_Wimplicit_function_declaration,
+	  warned = pedwarn (&richloc, OPT_Wimplicit_function_declaration,
 				  "implicit declaration of function %qE;"
 				  " did you mean %qs?",
 				  id, suggestion);
 	}
       else
-	warned = permerror_opt (loc, OPT_Wimplicit_function_declaration,
+	warned = pedwarn (loc, OPT_Wimplicit_function_declaration,
 				"implicit declaration of function %qE", id);
     }
   else if (const char *suggestion = hint.suggestion ())
@@ -3879,7 +3879,7 @@ implicitly_declare (location_t loc, tree functionid)
 	     then recycle the old declaration but with the new type.  */
 	  if (!C_DECL_IMPLICIT (decl))
 	    {
-	      implicit_decl_permerror (loc, functionid, decl);
+	      implicit_decl_warning (loc, functionid, decl);
 	      C_DECL_IMPLICIT (decl) = 1;
 	    }
 	  if (fndecl_built_in_p (decl))
@@ -3932,7 +3932,7 @@ implicitly_declare (location_t loc, tree functionid)
   DECL_EXTERNAL (decl) = 1;
   TREE_PUBLIC (decl) = 1;
   C_DECL_IMPLICIT (decl) = 1;
-  implicit_decl_permerror (loc, functionid, 0);
+  implicit_decl_warning (loc, functionid, 0);
   asmspec_tree = maybe_apply_renaming_pragma (decl, /*asmname=*/NULL);
   if (asmspec_tree)
     set_user_assembler_name (decl, TREE_STRING_POINTER (asmspec_tree));
-- 
2.50.1

