From 94023c51367708670cf9fb14863748e3096c9319 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 1 Jul 2025 03:48:16 -0300
Subject: [PATCH] Remove hardcoded RPATH and versioned SONAME from libisl

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index d652fa5..8503afe 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1229,7 +1229,7 @@ libisl_la_SOURCES = \
 	isl_yaml.h
 
 libisl_la_LIBADD = @MP_LIBS@
-libisl_la_LDFLAGS = -version-info @versioninfo@ \
+libisl_la_LDFLAGS = -avoid-version \
 	@MP_LDFLAGS@
 
 isl_test_LDFLAGS = @MP_LDFLAGS@
-- 
2.49.0

