From 1ab167c97c44585c4d0007463271bd904687f81c Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Thu, 2 Oct 2025 04:25:14 -0300
Subject: [PATCH 09/11] Fix missing stdint.h include when compiling host tools
 on OpenBSD

---
 gcc/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/Makefile.in b/gcc/Makefile.in
index 39faa0807..cae9541bb 100644
--- a/gcc/Makefile.in
+++ b/gcc/Makefile.in
@@ -3267,7 +3267,7 @@ GENGTYPE_OBJS = gengtype.o gengtype-parse.o gengtype-state.o \
   gengtype-lex.o errors.o
 
 gengtype-lex.o build/gengtype-lex.o : gengtype-lex.cc gengtype.h $(SYSTEM_H)
-CFLAGS-gengtype-lex.o += -DHOST_GENERATOR_FILE
+CFLAGS-gengtype-lex.o += -DHOST_GENERATOR_FILE -include stdint.h
 build/gengtype-lex.o: $(BCONFIG_H)
 
 gengtype-parse.o build/gengtype-parse.o : gengtype-parse.cc gengtype.h \
-- 
2.50.1

