From 99daacb155df91ad1a5d04588da567aa4e4573a5 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Tue, 31 Mar 2026 17:24:40 +0200
Subject: [PATCH] Support Darwin AArch64 host

---
 gcc/config.host | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config.host b/gcc/config.host
index 76689072..8950ccb8 100644
--- a/gcc/config.host
+++ b/gcc/config.host
@@ -181,7 +181,7 @@ case ${host} in
   i[34567]86-*-interix3*)
     host_xmake_file="${host_xmake_file} x-interix"
     ;;
-  i[34567]86-*-darwin* | x86_64-*-darwin*)
+  i[34567]86-*-darwin* | x86_64-*-darwin* | aarch64-*-darwin*)
     out_host_hook_obj="${out_host_hook_obj} host-i386-darwin.o"
     host_xmake_file="${host_xmake_file} i386/x-darwin"
     ;;
-- 
2.53.0

