From 1359d503eeefddef9e50dff9cd52bb01cd929694 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Sat, 28 Mar 2026 13:12:29 +0100
Subject: [PATCH] Fix definition of abort() on Windows

---
 gcc/system.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gcc/system.h b/gcc/system.h
index 654c85f..6eeddfc 100644
--- a/gcc/system.h
+++ b/gcc/system.h
@@ -23,6 +23,10 @@ Software Foundation, 59 Temple Place - Suite 330, Boston, MA
 #ifndef GCC_SYSTEM_H
 #define GCC_SYSTEM_H
 
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#endif
+
 /* We must include stdarg.h/varargs.h before stdio.h.  */
 #ifdef ANSI_PROTOTYPES
 #include <stdarg.h>
-- 
2.53.0

