From d404d22287b1bfa9fba7be7e6afc208e229a27b5 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Wed, 25 Mar 2026 05:16:38 +0100
Subject: [PATCH] Prevent libstdc++ from trying to implement math stubs

---
 libstdc++-v3/src/c++98/Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libstdc++-v3/src/c++98/Makefile.in b/libstdc++-v3/src/c++98/Makefile.in
index c3173bc91..248e0a342 100644
--- a/libstdc++-v3/src/c++98/Makefile.in
+++ b/libstdc++-v3/src/c++98/Makefile.in
@@ -83,8 +83,8 @@ am__objects_5 = bitmap_allocator.lo pool_allocator.lo mt_allocator.lo \
 	codecvt.lo complex_io.lo ctype.lo globals_io.lo hash_tr1.lo \
 	hashtable_tr1.lo ios.lo ios_failure.lo ios_init.lo \
 	ios_locale.lo list.lo locale.lo locale_init.lo \
-	locale_facets.lo localename.lo math_stubs_float.lo \
-	math_stubs_long_double.lo stdexcept.lo strstream.lo tree.lo \
+	locale_facets.lo localename.lo \
+	stdexcept.lo strstream.lo tree.lo \
 	istream.lo streambuf.lo valarray.lo $(am__objects_1) \
 	$(am__objects_4)
 am_libc__98convenience_la_OBJECTS = $(am__objects_5)
-- 
2.53.0

