From 80e8838fcd7b37fcd92c741f6cb432ff19819173 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Sat, 6 Sep 2025 22:58:49 -0300
Subject: [PATCH] Turn -Wincompatible-pointer-types back into an warning

---
 gcc/c/c-typeck.cc | 24 ++++++++----------------
 1 file changed, 8 insertions(+), 16 deletions(-)

diff --git a/gcc/c/c-typeck.cc b/gcc/c/c-typeck.cc
index b007c8ffe..e0b2819ca 100644
--- a/gcc/c/c-typeck.cc
+++ b/gcc/c/c-typeck.cc
@@ -6323,15 +6323,7 @@ build_conditional_expr (location_t colon_loc, tree ifexp, bool ifexp_bcp,
       else
 	{
 	  int qual = ENCODE_QUAL_ADDR_SPACE (as_common);
-	  diagnostic_t kind = DK_PERMERROR;
-	  if (!flag_isoc99)
-	    /* This downgrade to a warning ensures that -std=gnu89
-	       -pedantic-errors does not flag these mismatches between
-	       builtins as errors (as DK_PERMERROR would).  ISO C99
-	       and later do not have implicit function declarations,
-	       so the mismatch cannot occur naturally there.  */
-	    kind = bltin1 && bltin2 ? DK_WARNING : DK_PEDWARN;
-	  if (emit_diagnostic (kind, colon_loc, OPT_Wincompatible_pointer_types,
+	  if (emit_diagnostic (bltin1 && bltin2 ? DK_WARNING : DK_PEDWARN, colon_loc, OPT_Wincompatible_pointer_types,
 			       "pointer type mismatch "
 			       "in conditional expression"))
 	    {
@@ -8573,7 +8565,7 @@ convert_for_assignment (location_t location, location_t expr_loc, tree type,
 		gcc_rich_location richloc (expr_loc, &rhs_label,
 					   highlight_colors::actual);
 		warned
-		  = permerror_opt (&richloc, OPT_Wincompatible_pointer_types,
+		  = pedwarn (&richloc, OPT_Wincompatible_pointer_types,
 				   "passing argument %d of %qE from "
 				   "incompatible pointer type",
 				   parmnum, rname);
@@ -8584,13 +8576,13 @@ convert_for_assignment (location_t location, location_t expr_loc, tree type,
 	    case ic_assign:
 	      if (bltin)
 		warned
-		  = permerror_opt (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn (location, OPT_Wincompatible_pointer_types,
 				   "assignment to %e from pointer to "
 				   "%qD with incompatible type %e",
 				   &e_type, bltin, &e_rhstype);
 	      else
 		warned
-		  = permerror_opt (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn (location, OPT_Wincompatible_pointer_types,
 				   "assignment to %e from incompatible "
 				   "pointer type %e",
 				   &e_type, &e_rhstype);
@@ -8599,13 +8591,13 @@ convert_for_assignment (location_t location, location_t expr_loc, tree type,
 	    case ic_init_const:
 	      if (bltin)
 		warned
-		  = permerror_init (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn_init (location, OPT_Wincompatible_pointer_types,
 				    "initialization of %e from pointer to "
 				    "%qD with incompatible type %e",
 				    &e_type, bltin, &e_rhstype);
 	      else
 		warned
-		  = permerror_init (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn_init (location, OPT_Wincompatible_pointer_types,
 				    "initialization of %e from incompatible "
 				    "pointer type %e",
 				    &e_type, &e_rhstype);
@@ -8613,13 +8605,13 @@ convert_for_assignment (location_t location, location_t expr_loc, tree type,
 	    case ic_return:
 	      if (bltin)
 		warned
-		  = permerror_opt (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn (location, OPT_Wincompatible_pointer_types,
 				   "returning pointer to %qD of type %e from "
 				   "a function with incompatible type %e",
 				   bltin, &e_rhstype, &e_type);
 	      else
 		warned
-		  = permerror_opt (location, OPT_Wincompatible_pointer_types,
+		  = pedwarn (location, OPT_Wincompatible_pointer_types,
 				   "returning %e from a function with "
 				   "incompatible return type %e",
 				   &e_rhstype, &e_type);
-- 
2.50.1

