From 7f22709101d52d507d8c1ee2ef28181ccc5e547d Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Wed, 25 Mar 2026 23:03:05 +0100
Subject: [PATCH] Prevent libstdc++ from trying to implement math stubs

---
 libstdc++-v3/src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libstdc++-v3/src/Makefile.in b/libstdc++-v3/src/Makefile.in
index a385e4c6d..c1341ee94 100644
--- a/libstdc++-v3/src/Makefile.in
+++ b/libstdc++-v3/src/Makefile.in
@@ -123,7 +123,7 @@ am__objects_5 = atomic.lo bitmap_allocator.lo pool_allocator.lo \
 	hashtable_c++0x.lo hashtable_tr1.lo ios.lo ios_failure.lo \
 	ios_init.lo ios_locale.lo limits.lo list.lo debug_list.lo \
 	locale.lo locale_init.lo locale_facets.lo localename.lo \
-	math_stubs_float.lo math_stubs_long_double.lo stdexcept.lo \
+	stdexcept.lo \
 	strstream.lo system_error.lo tree.lo allocator-inst.lo \
 	concept-inst.lo fstream-inst.lo ext-inst.lo ios-inst.lo \
 	iostream-inst.lo istream-inst.lo istream.lo locale-inst.lo \
-- 
2.53.0

