From 3ad2ab34ce7de1ac91f360dff390d5faa5027541 Mon Sep 17 00:00:00 2001
From: Kartatz <105828205+Kartatz@users.noreply.github.com>
Date: Fri, 27 Mar 2026 03:20:19 +0100
Subject: [PATCH] Default to --enable-twoprocess for MinGW32

---
 fixincludes/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fixincludes/configure b/fixincludes/configure
index 4d445194..789b404d 100755
--- a/fixincludes/configure
+++ b/fixincludes/configure
@@ -2562,7 +2562,7 @@ fi
 else
   case $host in
 	i?86-*-msdosdjgpp* | \
-	i?86-*-mingw32* | \
+	*-mingw32* | \
 	*-*-beos* )
 		TARGET=twoprocess
 		;;
-- 
2.53.0

